src/decode.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "riscv_private.h"
+
+#define RV32I_OPCODE_LIST \
+    _(load)               \
+    _(op_imm)             \
+    _(auipc)              \
+    _(store)              \
+    _(op)                 \
+    _(lui)                \
+    _(branch)             \
+    _(jalr)               \
+    _(jal)                \
+    _(system)
+
+#define Zifencei_OPCODE_LIST _(misc_mem)
```

Comment:
Drop the macros such as `Zifencei_OPCODE_LIST` and `RV32A_OPCODE_LIST` in favor of macro `IIF`.